data/random-teams.js

```diff
@@ -1881,32 +1881,52 @@ class RandomTeams extends Dex.ModdedDex {
 		}
 
 		let items = [];
-		if (Array.isArray(setData.set.item) === true) {
+		if (Array.isArray(setData.set.item)) {
 			let randomItem = setData.set.item;
 			items.push(setData.itemVariants ? randomItem[setData.itemVariants] : this.sample(randomItem));
 		} else {
 			items.push(setData.set.item);
 		}
 
 		let abilities = [];
-		if (Array.isArray(setData.set.ability) === true) {
+		if (Array.isArray(setData.set.ability)) {
 			let randomAbility = setData.set.ability;
 			abilities.push(setData.abilityVariants ? randomAbility[setData.abilityVariants] : this.sample(randomAbility));
 		} else {
 			abilities.push(setData.set.ability);
 		}
 
+		let evs = {};
+		if (setData.set.evs) {
+			for (let ev in setData.set.evs) {
+				if (!setData.set.evs[ev]) evs[ev] = 0;
+				evs[ev] = setData.set.evs[ev];
+			}
+		} else {
+			evs = {hp: 84, atk: 84, def: 84, spa: 84, spd: 84, spe: 84};
+		}
+
+		let ivs = {};
+		if (setData.set.ivs) {
+			for (let iv in setData.set.ivs) {
+				if (!setData.set.ivs[iv]) ivs[iv] = 31;
+				ivs[iv] = setData.set.ivs[iv];
+			}
+		} else {
+			ivs = {hp: 31, atk: 31, def: 31, spa: 31, spd: 31, spe: 31};
+		}
```

Comment:
Or I think `let ivs = Object.assign({hp: 31, atk: 31, def: 31, spa: 31, spd: 31, spe: 31}, setData.set.ivs);` also works.

Reply:
That's unnecessary